### PR TITLE
Make gradle 7 or greater compatible version

### PR DESCRIPTION
**Pull request by ghost**
Changing compile for implementation as it has been deprecated in gradle 7.

### CODE REVIEW

**Comment by ghost** [comment]
Aah. I was just about to create mr for same changes for one of our products. @uklance Any hope of getting this merged soon?

**Comment by nosajsnikta** [comment]
@uklance as above...any chance of merging this very useful plugin?

File: java-monkey-patch/src/main/groovy/com/lazan/JavaMonkeyPatchPlugin.groovy [hunk 1]
```diff
@@ -46,7 +46,7 @@ class JavaMonkeyPatchPlugin implements Plugin<Project> {
                 dependencies {
                     monkeyPatchTransitive target
                     monkeyPatchNonTransitive target
-                    compileOnly(target) {
```
Comment (nosajsnikta) [inline java-monkey-patch/src/main/groovy/com/lazan/JavaMonkeyPatchPlugin.groovy:49]: Is it necessary to replace this, as `compileOnly` is supported by Gradle 7.x?

**Comment by gordonbuntting** [comment]
It would be useful if you could address the comment so this could be merged and released @AlvarVG .  Plugin seems broken with current Gradle versions.

**Pull request closed without merging**